api/get.js

```diff
@@ -0,0 +1,179 @@
+var _ = require('underscore'),
+    Q = require('q'),
+    glob = require('glob'),
+    fs = require('fs'),
+    path = require('path'),
+    request = require('request'),
+    config = require('config'),
+    
+    restify = require('restify'),
+    
+    overriders = require('./overriders'),
+    
+    folders = require('./folders'),
+    equals = require('./comparator');
+
+function Api(urlList) {
+    this.folder = folders.path;
+    this.urlList = urlList;
+}
+Api.prototype = {
+    handle: function(req, res, next) {
+        if (!this.urlList.hasOwnProperty(req.params.path)) {
+            res.send(404, 'Not handled!');
+            return next(new restify.errors.ResourceNotFoundError('Not Handled', '"' + req.params.path + '" is not defined by configuration'));
+        }
+        
+        var proxy = this.urlList[req.params.path];
+        overriders.save.folders(proxy, proxy.route);
+        this.folder = folders[proxy.name];
+        this.proxy(proxy, req, res, next)
+            .then(function() {
+                console.log('resolved', arguments);
+                return next();
+            })
+            .fail(function(error) {
+                console.error(error);
+                return next(error);
+            });
+    },
+    
+    
+    proxy: function(proxy, req, res, next) {
+        var url = proxy.url;
+        var patterns = [ path.join(folders[proxy.name], '*.req') ];
+        overriders.override.if.needed(patterns, proxy, this);
+        return Q.allSettled(_.map(patterns, function(pattern) {
```

Comment:
This whole block doesn't read too well. Let's document what it does, I can help with unit tests.